### PR TITLE
feat(panel): add content-top slot

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #8980

## Summary
Add a new `content-top` slot to the `panel` component.

### CODE REVIEW

File: packages/calcite-components/src/components/panel/panel.stories.ts [hunk 1]
```diff
@@ -440,3 +440,24 @@ export const withNoHeaderBorderBlockEnd_TestOnly = (): string =>
   html`<calcite-panel style="--calcite-panel-header-border-block-end:none;" height-scale="s" heading="My Panel"
     >Slotted content!</calcite-panel
   >`;
+
+export const contentTopSlot_TestOnly = (): string => html`
```
Comment (jcfranco) [inline packages/calcite-components/src/components/panel/panel.stories.ts:444]: New tests no longer need the `_TestOnly` suffixes. See updated [testing conventions](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/conventions/Testing.md#writing-stories).

File: packages/calcite-components/src/components/panel/panel.stories.ts [hunk 2]
```diff
@@ -440,3 +440,24 @@ export const withNoHeaderBorderBlockEnd_TestOnly = (): string =>
...
+        <calcite-action-group>
+          <calcite-action text="Add" icon="plus"> </calcite-action>
+          <calcite-action text="Save" icon="save"> </calcite-action>
+          <calcite-action text="Layers" icon="layers"> </calcite-action>
+        </calcite-action-group>
+      </calcite-action-bar>
+      <div slot="content-top" id="content-top">Slot for a content-top.</div>
+      <div slot="header-content">Header!</div>
+      <p>Slotted content!</p>
+      <p style="height: 400px">Hello first!</p>
```
Comment (jcfranco) [inline packages/calcite-components/src/components/panel/panel.stories.ts:457]: Is this the simplest test case or can we simplify by removing these paragraphs with inline styles?

File: packages/calcite-components/src/components/panel/panel.stories.ts [hunk 3]
```diff
@@ -440,3 +440,24 @@ export const withNoHeaderBorderBlockEnd_TestOnly = (): string =>
   html`<calcite-panel style="--calcite-panel-header-border-block-end:none;" height-scale="s" heading="My Panel"
     >Slotted content!</calcite-panel
   >`;
+
+export const contentTopSlot_TestOnly = (): string => html`
+  <div style="height: 300px; width: 400px; display: flex">
```
Comment (jcfranco) [inline packages/calcite-components/src/components/panel/panel.stories.ts:445]: Is this outer element needed for the `content-top` slot screenshot?

File: packages/calcite-components/src/components/panel/panel.scss [hunk 4]
```diff
@@ -18,6 +18,25 @@
 
 @import "../../assets/styles/header";
 
+:host([scale="s"]) .content-top {
+  padding: theme("spacing.2");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/panel/panel.scss:22]: For new styles, can we use the token CSS props vs `theme`?

**Review by jcfranco — APPROVED** [review]
Aside from a few comments, this LGTM! 🚀

File: packages/calcite-components/src/components/panel/panel.stories.ts (hunk 2 above)
Comment (Elijbet) [reply packages/calcite-components/src/components/panel/panel.stories.ts:457]: I can remove inline styles, but paragraphs are there to show the sticky property of `content-top` when scrolled.

File: packages/calcite-components/src/components/panel/panel.stories.ts (hunk 3 above)
Comment (Elijbet) [reply packages/calcite-components/src/components/panel/panel.stories.ts:445]: The outer element is there to restrict the height of the panel so we can see how scrolling of the slotted content affects the header (it should be sticky).

File: packages/calcite-components/src/components/panel/panel.tsx [hunk 5]
```diff
@@ -48,6 +48,7 @@ import { CSS, ICONS, SLOTS } from "./resources";
 /**
  * @slot - A slot for adding custom content.
  * @slot action-bar - A slot for adding a `calcite-action-bar` to the component.
+ * @slot content-top - A slot for adding the component's content header.
```
[the remark this answers is not in the document]
Comment (DitwanP) [reply packages/calcite-components/src/components/panel/panel.tsx:51]: Yeah I think the suggestion that @macandcheese made looks good, or something very close to it.

**Comment by driskull** [comment]
@Elijbet @jcfranco should these slots also be added to `flow-item` for consistency?

**Pull request merged**